main.cc

```diff
@@ -1639,6 +1644,67 @@ To start the scylla server proper, simply invoke as: scylla server (or just scyl
+            const qualified_name qualified_authenticator_name(auth::meta::AUTH_PACKAGE_NAME, cfg->authenticator());
+            const qualified_name qualified_role_manager_name(auth::meta::AUTH_PACKAGE_NAME, cfg->role_manager());
+
+            auth::service_config auth_config;
+            auth_config.authorizer_java_name = qualified_authorizer_name;
+            auth_config.authenticator_java_name = qualified_authenticator_name;
+            auth_config.role_manager_java_name = qualified_role_manager_name;
+
+            auth::service_config maintenance_auth_config;
+            maintenance_auth_config.authorizer_java_name = sstring{auth::allow_all_authorizer_name};
+            maintenance_auth_config.authenticator_java_name = sstring{auth::allow_all_authenticator_name};
+            maintenance_auth_config.role_manager_java_name = sstring{auth::maintenance_port_role_manager_name};
+
+            auth_service.start(perm_cache_config, std::ref(qp), std::ref(mm_notifier), std::ref(mm), auth_config, maintenance_port_enabled::no).get();
+            maintenance_auth_service.start(perm_cache_config, std::ref(qp), std::ref(mm_notifier), std::ref(mm), maintenance_auth_config, maintenance_port_enabled::yes).get();
+
+            auto start_auth_service = [&mm] (sharded<auth::service>& auth_service, std::any& stop_auth_service) {
+                supervisor::notify("starting auth service");
+                auth_service.invoke_on_all(&auth::service::start, std::ref(mm)).get();
+
+                stop_auth_service = defer_verbose_shutdown("auth service", [&auth_service] {
+                    auth_service.stop().get();
+                });
+            };
+
+            api::set_server_authorization_cache(ctx, auth_service).get();
+            auto stop_authorization_cache_api = defer_verbose_shutdown("authorization cache api", [&ctx] {
+                api::unset_server_authorization_cache(ctx).get();
+            });
+
+            scheduling_group_key_config cql_sg_stats_cfg = make_scheduling_group_key_config<cql_transport::cql_sg_stats>(maintenance_port_enabled::no);
+            scheduling_group_key_config maintenance_cql_sg_stats_cfg = make_scheduling_group_key_config<cql_transport::cql_sg_stats>(maintenance_port_enabled::yes);
+            cql_transport::controller cql_server_ctl(auth_service, mm_notifier, gossiper, qp, service_memory_limiter, sl_controller, lifecycle_notifier, *cfg, scheduling_group_key_create(cql_sg_stats_cfg).get0(), maintenance_port_enabled::no);
+            cql_transport::controller cql_maintenance_server_ctl(maintenance_auth_service, mm_notifier, gossiper, qp, service_memory_limiter, sl_controller, lifecycle_notifier, *cfg, scheduling_group_key_create(maintenance_cql_sg_stats_cfg).get0(), maintenance_port_enabled::yes);
+
+            auto start_cql = [&dbcfg] (cql_transport::controller& controller, std::any& stop_cql) {
```

Comment:
Code movements and refactoring (such as extraction of `start_cql` function) should be in a separate commit than the new stuff.